frontend/src/components/SignUp.js

```diff
@@ -33,14 +32,42 @@ function Copyright(props) {
 
 const theme = createTheme();
 
+const axios = require("axios").default;
+
+const FeedbackField = (props) => {
+  const response = props.response;
+
+  if (response === null) return;
+
+  if (response.status === 201) {
+    return (
+      <Box marginBottom={2}>
+        <Alert severity="success">User has been created</Alert>
+      </Box>
+    );
+  } else {
+    return (
+      <Box marginBottom={2}>
+        <Alert severity="error">{response.data.error}</Alert>
+      </Box>
+    );
+  }
+};
```

Comment:
```suggestion
   if (response != null){

    var response_box = (
      <Box marginBottom={2}>
        <Alert severity="error">{response.data.error}</Alert>
      </Box>
    )

    switch(response.status){
      case 201:
        response_box = (
        <Box marginBottom={2}>
          <Alert severity="success">User has been created</Alert>
        </Box>
        )
        break; 
      case 422:
        response_box = (
        <Box marginBottom={2}>
          <Alert severity="error">Some fields are missing</Alert>
        </Box>)
    }
    
    return response_box
  }
};
```

Reply:
We also have to handle situation when a user passes no password/username (status 422).
In such case, we could use switch, to avoid having multiple if statements

Reply:
What about frontend validation? So you will not allow a user to send a form without those fields.